grid_map_visualization/src/GridMapVisualization.cpp

```diff
@@ -6,165 +6,170 @@
-                    visualizationsParameter_.c_str(), config[j].getType());
-          return false;
-        }
-
-        if (!config[j].hasMember("name")
-            || config[i]["name"].getType() != XmlRpc::XmlRpcValue::TypeString
-            || config[j]["name"].getType() != XmlRpc::XmlRpcValue::TypeString) {
-          ROS_ERROR("%s: Visualizations names must be strings, but they are XmlRpcTypes:%d and %d",
-                    visualizationsParameter_.c_str(), config[i].getType(), config[j].getType());
-          return false;
-        }
-
-        std::string namei = config[i]["name"];
-        std::string namej = config[j]["name"];
-        if (namei == namej) {
-          ROS_ERROR("%s: A visualization with the name '%s' already exists.",
-                    visualizationsParameter_.c_str(), namei.c_str());
-          return false;
-        }
+      RCLCPP_ERROR(
+        node_->get_logger(),
+        "%s: A visualization with the name '%s' already exists.",
+        visualizationsParameter_.c_str(), name.c_str());
+      return false;
+    }
+
+    node_->declare_parameter(name + ".type");
+    try {
+      if (!node_->get_parameter(name + ".type", type)) {
+        RCLCPP_ERROR(
+          node_->get_logger(),
+          "%s: Could not add a visualization because no type was given",
+          name.c_str());
+        return false;
       }
+    } catch (const rclcpp::ParameterTypeException & e) {
+      RCLCPP_ERROR(
+        node_->get_logger(),
+        "Could not add %s visualization, because the %s.type parameter is not a string.",
+        name.c_str(), name.c_str());
+      return false;
     }
 
     // Make sure the visualization has a valid type.
-    if (!factory_.isValidType(config[i]["type"])) {
-      ROS_ERROR("Could not find visualization of type '%s'.", std::string(config[i]["type"]).c_str());
+    if (!factory_->isValidType(type)) {
+      RCLCPP_ERROR(
+        node_->get_logger(),
+        "Could not add %s visualization, no visualization of type '%s' found.",
+        name.c_str(), type.c_str());
       return false;
     }
-  }
 
-  for (int i = 0; i < config.size(); ++i) {
-    std::string type = config[i]["type"];
-    std::string name = config[i]["name"];
-    auto visualization = factory_.getInstance(type, name);
-    visualization->readParameters(config[i]);
+    auto visualization = factory_->getInstance(type, name);
+    visualization->readParameters();
     visualizations_.push_back(visualization);
-    ROS_INFO("%s: Configured visualization of type '%s' with name '%s'.",
-             visualizationsParameter_.c_str(), type.c_str(), name.c_str());
+    RCLCPP_INFO(
+      node_->get_logger(), "%s: Configured visualization of type '%s' with name '%s'.",
+      visualizationsParameter_.c_str(), type.c_str(), name.c_str());
   }
-
   return true;
 }
 
 bool GridMapVisualization::initialize()
 {
-  for (auto& visualization : visualizations_) {
+  for (auto visualization : visualizations_) {
```

Comment:
Below you did not remove the `&`

Reply:
Will reverted this change, removed while debugging and forgot to revert it back, my bad.